core/src/main/scala/neotypes/DeferredQuery.scala

```diff
@@ -323,6 +325,15 @@ final case class DeferredQuery[T] private[neotypes] (query: String, params: Map[
     */
   def withParams(params: Map[String, QueryParam]): DeferredQuery[T] =
     this.copy(params = this.params ++ params)
+
+  private[neotypes] def withParameterPrefix(prefix: String): DeferredQuery[T] = {
```

Comment:
Having this method in this class feels wrong. But I understand that otherwise, we would need to duplicate logic or move this work that right now is done at compile time _(which is great)_ to runtime _(which would be worse to users)_.

What about if we move both the `query[T]` method as well as this one to a helper object?
Especially because once the final query is built, these param locations are actually unnecessary, right?

Reply:
I actually disagree with that. It deals with the internal structure of the query so is the correct place for it if you ask me. If you want me to move it I can do that though. Just tell me where :)

Reply:
> No, they need to be kept so that parameter names can be prefixed if the query later gets nested in another one

Yeah, that is what I tried to mean _"final query"_.

The reason I do not like it here is that we do not want to embed a `DefferedQuery` inside a `DefferedQueryBuilder`, we want to embed a `DefferedQueryBuilder` inside another one; that is why having to call `query` for the builder in order to embed it in another builder feels weird for me. 
Thus, I think it would be good to move both this method as well as the loop inside `query` to a helper object. But do not ask me for naming, I suck at that 😅

Does that make more sense to you? Or what do you think?

Reply:
> The reason I do not like it here is that we do not want to embed a DefferedQuery inside a DefferedQueryBuilder, we want to embed a DefferedQueryBuilder inside another one...

I'm not sure that's possible. At least, I wouldn't know how to do it.

Reply:
Uhm, let's just leave it as it is right and I will try to play with the code later.
In any case, I am no even sure if my idea makes sense since I haven't tried to code it :)